clevertap-core/src/main/java/com/clevertap/android/sdk/network/api/DefineTemplatesRequestBody.kt

```diff
@@ -0,0 +1,44 @@
+package com.clevertap.android.sdk.network.api
+
+import com.clevertap.android.sdk.inapp.customtemplates.CustomTemplate
+import com.clevertap.android.sdk.utils.putObject
+import org.json.JSONArray
+import org.json.JSONObject
+
+internal class DefineTemplatesRequestBody(header: JSONObject, templates: Collection<CustomTemplate>) {
+
+    val jsonArray = JSONArray().apply {
+        put(header)
+        put(templates.toJSON())
+    }
+
+    override fun toString(): String = jsonArray.toString()
+}
+
+private fun Collection<CustomTemplate>.toJSON(): JSONObject {
+    val templates = this
+
+    return JSONObject().apply {
```

Comment:
lint might warn for constants.

Reply:
It does not complain on my IDE. I think directly using the values in this case makes the whole block more readable, but if we must use constants for those cases I can change it.